src/examples/ExamplesPanel.ts

```diff
@@ -82,8 +83,8 @@ export class ExamplesPlanel {
                     selectedFolder[0].fsPath,
                     message.name
                   );
-                  ensureDir(resultFolder).then(() => {
-                    utils.copyFromSrcProject(
+                  ensureDir(resultFolder).then(async () => {
+                    await utils.copyFromSrcProject(
```

Comment:
Will this block the UI / vscode extension/ extension host if the example is significantly big .... say 200mb+ ?

Reply:
This should not block the UI since there is no await before ensure. Should remove the await from the utils.copyFromSrcProject function. About the waiting time, if the user has built within IDF_PATH/examples then probably. We could ignore build folder in this function to mitigate long time copies.